src/stcal/jump/jump.py

```diff
@@ -1118,12 +908,319 @@ def find_faint_extended(
+
+    #  get the contours that are above the minimum size
+    bigcontours = [con for con in contours if cv.contourArea(con) > jump_data.extend_min_area]
+    return bigcontours 
+
+
+def diff_meddiff_int(intg, median_diffs, sigma, first_diffs_masked):
+    """
+    Computes the SNR ratio of each difference.
+
+    Parameters
+    ----------
+    intg : int
+        Current intregration
+
+    median_diffs : ndarray
+        Median of differences in integration
+
+    sigma : ndarray
+        Weighting.
+
+    first_diffs_masked : ndarray
+        Masked first differences.
+
+    Return
+    ------
+    median_diffs : ndarray
+        Median of first differences
+
+    ratio : ndarray
+        SNR ratio
+    """
+    if intg > 0:
+        e_jump = first_diffs_masked[intg] - median_diffs[np.newaxis, :, :]
+
+        # SNR ratio of each diff.
+        ratio = np.abs(e_jump) / sigma[np.newaxis, :, :]
+    else:
+        # The difference from the median difference for each group
+        e_jump = first_diffs_masked[intg] - median_diffs[np.newaxis, :, :]
+
+        # SNR ratio of each diff.
+        ratio = np.abs(e_jump) / sigma[np.newaxis, :, :]
+        median_diffs = np.nanmedian(first_diffs_masked, axis=(0, 1))
+
+    return median_diffs, ratio
+
+
+def diff_meddiff_grp(intg, grp, median, stddev, first_diffs_masked):
+    """
+    Find the median difference group.
+
+    Parameters
+    ----------
+    intg : int
+        Current intregration
+
+    grp : int
+        Current group
+
+    median : float
+        Median computed during sigma clipping.
+
+    stddev : float
+        Standard deviation computed during sigma clipping.
+
+    first_diffs_masked : ndarray
+        Masked first differences.
+
+    Return
+    ------
+    median_diffs : ndarray
+        Median of first differences
+
+    ratio : ndarray
+        SNR ratio
+    """
+    median_diffs = median[grp - 1]
+    sigma = stddev[grp - 1]
+
+    # The difference from the median difference for each group
+    e_jump = first_diffs_masked[intg] - median_diffs[np.newaxis, :, :]
+
+    # SNR ratio of each diff.
+    ratio = np.abs(e_jump) / sigma[np.newaxis, :, :]
+
+    return median_diffs, ratio
+
+
+def nan_invalid_data(data, gdq, jump_data):
+    """
+    Mark flagged data as invalid by setting the science data to NaN.
+
+    Parameters
+    ----------
+    data : ndarray
+        Science data 4D float
+
+    gdq : ndarray
+        Group DQ 4D uint8
+
+    jump_data : JumpData
+        Class containing parameters and methods to detect jumps.
+
+    Return
+    ------
+    data : ndarray
+        NaN'd cience data 4D float
+    """
+    jump_dnu_flag = jump_data.fl_jump + jump_data.fl_dnu
+    sat_dnu_flag = jump_data.fl_sat + jump_data.fl_dnu
+    data[gdq == jump_dnu_flag] = np.nan
+    data[gdq == sat_dnu_flag] = np.nan
+    data[gdq == jump_data.fl_sat] = np.nan
+    data[gdq == jump_data.fl_jump] = np.nan
+    data[gdq == jump_data.fl_dnu] = np.nan
```

Comment:
```suggestion
    dnu_jump_sat = jump_data.fl_jump | jump_data.fl_dnu | jump_data.fl_sat
    data[gdq & dnu_jump_sat != 0] = np.nan
```
This function is now so limited that I suggest removing it and moving these two lines to where the function is called.

Reply:
Your recommended changes are not logically equivalent to the original code.  For example `0x03==0x02` is false, but `0x03 & 0x02` is true.  The original code isn't looking for any pixel where `dnu`, `jump`, and `sat` are some of the flags are set, but where those flags are the only flags set.

Reply:
You're right.  It does seem strange to me that this would be the intended behavior.  If these flags are set, but also a hot pixel flag, then we don't want to set the value to nan?